### DISCUSSION TITLE
File path for Docker socket on Synology may not be accurate

### DISCUSSION THREAD

**Discussion by JonMHudgins** [nextcloud/all-in-one#2510] [Ideas] [answered]
In the readme documentation, under "How to run AIO on Synology DSM" (found here: https://github.com/nextcloud/all-in-one#how-to-run-aio-on-synology-dsm) it is stated that in order to run the command on Synology DSM you need to change to change a few things. The section states: 

> ...instead of using `--volume /var/run/docker.sock:/var/run/docker.sock:ro`, you need to use `--volume /volume1/docker/docker.sock:/var/run/docker.sock:ro` to run it. You also need to add `--env WATCHTOWER_DOCKER_SOCKET_PATH="/volume1/docker/docker.sock"` to the docker run command of the mastercontainer...

I attempted to build my command with those changes, and I wound up getting the error:

> docker: Error response from daemon: Bind mount failed: '/volume1/docker/docker.sock' does not exists.

I thought that was odd, so I Googled about a bit and couldn't find a clear answer on my specific use case as to where the socket path was located. I finally determined that other Synology users were using '/var/run/docker.sock' as the path, and I modified my command accordingly and it worked.

To try to make a long story short, I'm not sure how widespread the differences are between different Synology models and DSM versions, but here are the specifics to my setup, in hopes that they may help others who are scratching their heads as I was.

> Model: Synology DS920+
> CPU: Intel Celeron J4125 (4 cores)
> RAM: 8GB (4+4)
> DSM Version: DSM 7.1.1-42962 Update 5
> Docker Version: 20.10.3-1308
> Docker Socket: /var/run/docker.sock

The full command that worked for me :
_(important note - I already had reverse proxy set up with Synology's built-in reverse proxy, so the command I used below is from the Reverse Proxy page found here https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#2-use-this-startup-command)_

> `sudo docker run \`
> `--sig-proxy=false \`
> `--name nextcloud-aio-mastercontainer \`
> `--restart always \`
> `--publish 8080:8080 \`
> `--env APACHE_PORT=7401 \`
> `--env APACHE_IP_BINDING=0.0.0.0 \`
> `--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \`
> `--volume /var/run/docker.sock:/var/run/docker.sock:ro \`
> `--env WATCHTOWER_DOCKER_SOCKET_PATH="/var/run/docker.sock" \`
> `nextcloud/all-in-one:latest`

**Comment by szaimen** [comment]
The problem is that the docker socket on synology apparently may be stored in the given directory or in the one that you used. But I aggree we should mention this in the docs. Would you mind creating a PR that adds this hint?

**Comment by szaimen** [comment, marked as answer]
Adjusted with https://github.com/nextcloud/all-in-one/commit/1f7f675454a12cb2772fb8783ba0c9b56deb6fc6
